var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -541,6 +548,16 @@ def define_enable(suffix, value=None):
                 define_trilinos_enable('DataTransferKit', True),
             ])
 
+        if '+aprepro' in spec:
+            options.extend([
+                define_trilinos_enable('SEACAS', True),
+                define_trilinos_enable('SEACASAprepro', True),
+            ])
+        else:
+            options.extend([
+                define_trilinos_enable('SEACASAprepro', False),
+            ])
```

Comment:
This type of logic should **not** exist in this package.py file.  This the type of logic that TriBITS handles automatically.

Reply:
The `SEACASAprepro` options need to be there so that the variants are consistent with what's actually installed... the `define_trilinos_enable('SEACAS', True)` could probably be dropped though. I think this one (and several of the similar ones) can be written:
```suggestion
        options.append(define_trilinos_enable('SEACAS', 'aprepro'))
```

Reply:
The issue with variants is that they have to be determined at concretization time in Spack to ensure they're compatible with all the other software in a spec (or set of specs in an environment). We can't just defer to tribits logic. The closest we could possibly do would be to parse the tribits dependencies for all the different versions of trilinos we want to support, then convert those into the appropriate variants and conflicts...

Reply:
> The issue with variants is that they have to be determined at concretization time in Spack to ensure they're compatible with all the other software in a spec (or set of specs in an environment).

I see, that makes sense.

> The closest we could possibly do would be to parse the tribits dependencies for all the different versions of trilinos we want to support, then convert those into the appropriate variants and conflicts...

Yes, that is what should be done.  Given the generated file XML file [TrilinosDependencies.xml](https://tribitspub.github.io/TriBITS/users_guide/index.html#project-packagedependencies-xml) (which could just be returned as a string to STDOUT as called inside Trilinos/package.py), Python code in Trilinos/package.py could process this and automatically generate the package enable/disable variants and provide the conflicts list (which will take some graph algorithms to give Spack what it wants and may need to be implemented carefully to be performant).